### PR TITLE
Fixes #3089: Dismiss pop-over on budget action

### LINKED ISSUE #3089: [Bug]: clicking on 'Copy last month's budget' should hide the modal after applying
### Verified issue does not already exist?

- [X] I have searched and found no existing issue
- [X] I will be providing steps how to reproduce the bug (in most cases this will also mean uploading a demo budget file)

### What happened?

Not sure if bug or feature.

On a budget of a certain month, clicking on `Copy last month's bugdet` (or on any option in that dropdown list) should hide the modal once the the action is completed.

Right now, it doesn't:
[Screencast from 2024-07-17 23-18-51.webm](https://github.com/user-attachments/assets/700c94c8-743c-4450-97ae-14b3a01a4158)

### Where are you hosting Actual?

Pikapods

### What browsers are you seeing the problem on?

Firefox

### Operating System

Linux

### PR DESCRIPTION
**Pull request by sleepyfran**
The component was not actually calling the dismissal of the pop-over. I've created a little utility that takes whatever arguments `onBudgetAction` takes, calls that function with those params and then dismisses the pop-over.

This fixes #3089 

## Little demo

Kapture 2024-07-18 at 00 45 28

### CODE REVIEW

**Comment by joel-jeremy** [comment]
Thank you for creating a PR! Can you also apply the same to the ReportComponents file?

**Comment by sleepyfran** [comment]
> Thank you for creating a PR! Can you also apply the same to the ReportComponents file?

From what I can see all of the paths that have any interactions with budget actions and pop-overs already do the manual close (verified and it's working). I could refactor them to be similar to what I applied in here if you'd like :)

**Comment by joel-jeremy** [comment]
> From what I can see all of the paths that have any interactions with budget actions and pop-overs already do the manual close (verified and it's working). I could refactor them to be similar to what I applied in here if you'd like :)

I see. Would you mind updating them to be similar?

**Comment by sleepyfran** [comment]
> > From what I can see all of the paths that have any interactions with budget actions and pop-overs already do the manual close (verified and it's working). I could refactor them to be similar to what I applied in here if you'd like :)
> 
> I see. Would you mind updating them to be similar?

Sure, done in the latest commit 😃 Also verified it's working:
Kapture 2024-07-18 at 20 39 25

Let me know what you think, I've unified both the regular menu and the balance menu on the same callback.

File: packages/desktop-client/src/components/budget/rollover/RolloverComponents.tsx [hunk 1]
```diff
@@ -221,7 +226,7 @@ export const ExpenseCategoryMonth = memo(function ExpenseCategoryMonth({
             >
               <BudgetMenu
                 onCopyLastMonthAverage={() => {
-                  onBudgetAction?.(month, 'copy-single-last', {
+                  onPopoverAction(month, 'copy-single-last', {
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/rollover/RolloverComponents.tsx:229]:
```suggestion
                  onMenuAction(month, 'copy-single-last', {
```

File: packages/desktop-client/src/components/budget/rollover/RolloverComponents.tsx [hunk 2]
```diff
@@ -234,12 +239,12 @@ export const ExpenseCategoryMonth = memo(function ExpenseCategoryMonth({
                     return;
                   }
 
-                  onBudgetAction?.(month, `set-single-${numberOfMonths}-avg`, {
+                  onPopoverAction(month, `set-single-${numberOfMonths}-avg`, {
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/rollover/RolloverComponents.tsx:242]:
```suggestion
                  onMenuAction(month, `set-single-${numberOfMonths}-avg`, {
```

File: packages/desktop-client/src/components/budget/rollover/RolloverComponents.tsx [hunk 3]
```diff
@@ -234,12 +239,12 @@ export const ExpenseCategoryMonth = memo(function ExpenseCategoryMonth({
...
                     category: category.id,
                   });
                 }}
                 onApplyBudgetTemplate={() => {
-                  onBudgetAction?.(month, 'apply-single-category-template', {
+                  onPopoverAction(month, 'apply-single-category-template', {
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/rollover/RolloverComponents.tsx:247]:
```suggestion
                  onMenuAction(month, 'apply-single-category-template', {
```

File: packages/desktop-client/src/components/budget/rollover/RolloverComponents.tsx [hunk 4]
```diff
@@ -158,6 +158,11 @@ export const ExpenseCategoryMonth = memo(function ExpenseCategoryMonth({
   const [balanceMenuOpen, setBalanceMenuOpen] = useState(false);
   const [hover, setHover] = useState(false);
 
+  const onPopoverAction = (...args: Parameters<typeof onBudgetAction>) => {
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/rollover/RolloverComponents.tsx:161]:
```suggestion
  const onMenuAction = (...args: Parameters<typeof onBudgetAction>) => {
```

**Review by joel-jeremy — COMMENTED** [review]
LGTM! Just one last comment to rename the function just to be consistent on the two components.

**Comment by sleepyfran** [comment]
> LGTM! Just one last comment to rename the function just to be consistent on the two components.

Renamed! Thanks a lot 😃

**Comment by joel-jeremy** [comment]
@sleepyfran Thank you for your contribution!

**Review by joel-jeremy — APPROVED** [review]

**Pull request merged**